src/cfnlint/rules/resources/codepipeline/CodepipelineStageActions.py

```diff
@@ -159,15 +161,30 @@ def check_version(self, action, path):
         """Check that action type version is valid."""
         matches = []
 
+        REGEX_VERSION_STRING = re.compile(r'^[0-9A-Za-z_-]+$')
+        LENGTH_MIN = 1
+        LENGTH_MAX = 9
+
         version = action.get('ActionTypeId', {}).get('Version')
         if isinstance(version, dict):
             self.logger.debug('Unable to validate version when an object is used.  Skipping')
-        elif version != '1':
-            message = 'For all currently supported action types, the only valid version string is "1".'
-            matches.append(RuleMatch(
-                path + ['ActionTypeId', 'Version'],
-                message
-            ))
+        elif isinstance(version, (six.string_types)):
+            if len(version) < LENGTH_MIN:
```

Comment:
This makes sense to me.  Only thing would just be simplification question?
Could we have done `if not LENGTH_MIN <= len(version) <= LENGTH_MAX:` and then just squash the message to include the min and the max?

Reply:
Sure! The other thing I noticed is that if you don't hit min length, you don't comply with the regex, so that could get thrown as well.